rareapi/views/reaction.py

```diff
@@ -0,0 +1,69 @@
+"""View module for handling requests about reactions"""
+from django.http import HttpResponseServerError
+from django.core.exceptions import ValidationError
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from rest_framework import serializers
+from rest_framework import status
+from rest_framework.decorators import action
+from rareapi.models import Reaction, PostReaction, Post, RareUser
+
+class Reactions(ViewSet):
+    """rare reactions"""
+
+    def list(self, request):
+        """Handle GET requests to reactions resource
+
+        Returns:
+            Response -- JSON serialized list of reactions
+        """
+        reactions = Reaction.objects.all()
+
+        post_id = self.request.query_params.get('post_id', None)
+        if post_id is not None:
+            for reaction in reactions:
+                post_reactions = PostReaction.objects.filter(post_id=post_id, reaction=reaction)
+                reaction.count = len(post_reactions)
+
+
+        serializer = ReactionSerializer(
+            reactions, many=True, context={'request': request})
+        return Response(serializer.data)
+
+    @action(methods=['post'], detail=True)
+    def react(self, request, pk=None):
+        """Managing users reacting to posts"""
+
+        if request.method == "POST":
+            post = Post.objects.get(pk=request.data["post_id"])
+            rareuser = RareUser.objects.get(user=request.auth.user)
+            reaction = Reaction.objects.get(pk=pk)
+
+            try:
+                # Determine if the user already reacted
+                like = PostReaction.objects.get(
+                    post=post, user=rareuser, reaction=reaction)
+                like.delete()
+                return Response({}, status=status.HTTP_204_NO_CONTENT)
+                
+            except PostReaction.DoesNotExist:
+                # The user is not signed up.
```

Comment:
This comment should probably say something like " the user has not reacted". Other than that this looks amazing!